File: app/src/main/java/com/anthonysottile/kenken/ui/UIConstants.java
package com.anthonysottile.kenken.ui;

import android.graphics.Color;
import android.graphics.Paint;

public final class UIConstants {

    public static final int MinGameSize = 4;
    public static final int MaxGameSize = 9;
    public static final int GameSizes = UIConstants.MaxGameSize - UIConstants.MinGameSize + 1;

    public static final int BorderWidth = 2;

    public static final int StatisticsOneIndent = 20;

    public static final int CandidatesTextColor = Color.rgb(0x77, 0x77, 0x77);

    private static Paint gridColor = null;

    public static Paint GetGridColor() {
        if (UIConstants.gridColor == null) {
            UIConstants.gridColor = new Paint();
            UIConstants.gridColor.setColor(Color.rgb(0xCC, 0xCC, 0xCC));
        }
        return UIConstants.gridColor;
    }

    private static Paint cageColor = null;

    public static Paint GetCageColor() {
        if (UIConstants.cageColor == null) {
            UIConstants.cageColor = new Paint();
            UIConstants.cageColor.setColor(Color.rgb(0x00, 0x00, 0x00));
        }
        return UIConstants.cageColor;
    }

    private UIConstants() {
    }
}

File: app/src/main/java/com/anthonysottile/kenken/ui/UIConstants.kt
package com.anthonysottile.kenken.ui

import android.graphics.Color
import android.graphics.Paint

object UIConstants {

    const val MinGameSize = 4
    const val MaxGameSize = 9
    const val GameSizes = UIConstants.MaxGameSize - UIConstants.MinGameSize + 1

    const val BorderWidth = 2

    const val StatisticsOneIndent = 20

    private var gridColor: Paint? = null

    private var cageColor: Paint? = null

    fun getGridColor(): Paint {
        if (UIConstants.gridColor == null) {
            UIConstants.gridColor = Paint()
            UIConstants.gridColor!!.color = Color.rgb(0xCC, 0xCC, 0xCC)
        }
        return UIConstants.gridColor!!
    }

    fun getCageColor(): Paint {
        if (UIConstants.cageColor == null) {
            UIConstants.cageColor = Paint()
            UIConstants.cageColor!!.color = Color.rgb(0x00, 0x00, 0x00)
        }
        return UIConstants.cageColor!!
    }
}
